Add support symfony 5

franmomu opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

I removed some test that looks like were added in https://github.com/sonata-project/SonataCoreBundle/pull/324 for Symfony 3.x and FQCN compatibilities.

<!-- Describe your Pull Request content here -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 1.x is for everything backwards compatible, like patches, features and deprecation notices
    - master is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/form-extensions/blob/1.x/CONTRIBUTING.md#the-base-branch
-->
I am targeting this branch, because these changes respect BC.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
    This will end up on https://github.com/sonata-project/form-extensions/releases,
    please keep it short and clear and to the point
-->

<!-- 
    If you are updating something that doesn't require
    a release, you can delete the whole Changelog section.
    (eg. update to docs, tests)
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Added
- Added Support for Symfony 5 packages
```

<!--
    If this is a work in progress, uncomment this section.
    You can add as many tasks as you want.
    If some are not relevant, just remove them.
    
    ## To do
    
    - [ ] Update the tests
    - [ ] Update the documentation
    - [ ] Add an upgrade note
-->

core23 commented:
Can you check the phpstan error @franmomu ?

This could help
https://github.com/sonata-project/form-extensions/pull/52/files#diff-5d8c276c617696f9eb6923af5b12a903

franmomu commented:
> Can you check the phpstan error @franmomu ?
> 
> This could help
> https://github.com/sonata-project/form-extensions/pull/52/files#diff-5d8c276c617696f9eb6923af5b12a903

The thing is that [IntlDateFormatter from Symfony doesn't implement formObject](https://github.com/symfony/symfony/blob/4.3/src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php#L238).

But there is a problem I've just realised that I'm trying to figure it out with Symfony 5. [BaseDatePickerType](https://github.com/sonata-project/form-extensions/blob/1.x/src/Type/BasePickerType.php) has `Symfony\Component\Translation\TranslatorInterface` as an argument, in Symfony 5 `Symfony\Contracts\Translation\TranslatorInterface` should be used instead, but the problem is that this interface doesn't have the `getLocale` method, so it should be fetched from the request.

franmomu commented:
So what I've done:
Move the StubTranslator and bump to php 7.2 [like CoreBundle](https://github.com/sonata-project/SonataCoreBundle/pull/723).

With `BasePickerType`, because there are two Interfaces to support, I removed the type hinting and add an internal check. The thing about these interfaces is that the current `Symfony\Component\Translation\TranslatorInterface` has a `getLocale` method and `Symfony\Contracts\Translation\TranslatorInterface` doesn't, so I added the third optional parameter [RequestStack to get the locale from there](https://github.com/symfony/symfony/issues/29530#issuecomment-445470461).

This check is to make sure that if you use the new interface (without `getLocale`), you should pass the RequestStack:

```php
if ($translator instanceof TranslatorInterface && null === $requestStack) {
    throw new \InvalidArgumentException(sprintf(
        'Argument 3 should be an instance of %s',
        RequestStack::class
    ));
}
```

I'll take a look at phpstan

phansys commented on tests/Type/DatePickerTypeTest.php:
```diff
@@ -85,13 +63,37 @@ public function testSubmitValidData(): void
+    /**
+     * @return MockObject|TranslatorInterface|LegacyTranslatorInterface\
+     */
+    private function getTranslatorMock()
```
```suggestion
    private function getTranslatorMock(): MockObject
```

phansys commented on tests/Type/DateTimePickerTypeTest.php:
```diff
@@ -105,13 +84,41 @@ public function testSubmitMatchingDateFormat(): void
+    /**
+     * @return MockObject|TranslatorInterface|LegacyTranslatorInterface\
+     */
+    private function getTranslatorMock()
```
```suggestion
    private function getTranslatorMock(): MockObject
```

greg0ire commented:
Thanks @franmomu !
